test/MarketMaker.js

```diff
@@ -108,81 +98,71 @@ contract('marketMaker', function(accounts) {
-      accounts[1], {from: accounts[0]});
+    mmInitialBalance = 20000000;
+    totalTokenSupply = 100;
+    rate = totalTokenSupply / web3.fromWei(mmInitialBalance+10000000, 'ether');
+    crowdale = await help.simulateCrowdsale(rate, [totalTokenSupply], accounts, 1);
+    token = LifToken.at( await crowdale.token.call());
+    mm = LifMarketMaker.at( await crowdale.marketMaker.call());
+    await mm.calculateDistributionPeriods({from: accounts[4]});
 
-    await mm.fund({value: web3.toWei(8, 'ether'), from: accounts[0]});
-    await mm.calculateDistributionPeriods({from: accounts[5]});
+    assert.equal(24, parseInt(await mm.totalPeriods.call()));
 
-    assert.equal(0, parseInt(web3.eth.getBalance(token.address)));
-    assert.equal(startBlock, parseInt(await mm.startBlock()));
-    assert.equal(blocksPerPeriod, parseInt(await mm.blocksPerPeriod()));
-    assert.equal(accounts[1], parseInt(await mm.foundationAddr()));
+    startTimestamp = parseInt(await mm.startTimestamp.call());
 
-    await help.waitToBlock(startBlock);
+    await increaseTimeTestRPCTo(startTimestamp);
     assert.equal(0, parseInt(await mm.getCurrentPeriodIndex()));
-    await help.waitToBlock(web3.eth.blockNumber+blocksPerPeriod);
+    await increaseTimeTestRPCTo(startTimestamp+duration.days(30));
     assert.equal(1, parseInt(await mm.getCurrentPeriodIndex()));
     await mm.pause({from: accounts[0]});
-    await help.waitToBlock(web3.eth.blockNumber+blocksPerPeriod*3);
+    await increaseTimeTestRPC(duration.days(30)*3);
     await mm.unpause({from: accounts[0]});
     assert.equal(1, parseInt(await mm.getCurrentPeriodIndex()));
-    await help.waitToBlock(web3.eth.blockNumber+blocksPerPeriod);
+    await increaseTimeTestRPC(duration.days(30));
     assert.equal(2, parseInt(await mm.getCurrentPeriodIndex()));
-    await help.waitToBlock(web3.eth.blockNumber+blocksPerPeriod);
+    await increaseTimeTestRPC(duration.days(30));
     assert.equal(3, parseInt(await mm.getCurrentPeriodIndex()));
     await mm.pause({from: accounts[0]});
-    await help.waitToBlock(web3.eth.blockNumber+blocksPerPeriod*2);
+    await increaseTimeTestRPC(duration.days(30)*2);
     await mm.unpause({from: accounts[0]});
-    await help.waitToBlock(web3.eth.blockNumber+blocksPerPeriod);
+    await increaseTimeTestRPC(duration.days(30));
     assert.equal(4, parseInt(await mm.getCurrentPeriodIndex()));
   });
 
   const periods = 24;
-  const tokenTotalSupply = 100;
+  const tokenTotalSupply = 3000;
   let customerAddressIndex = 1;
 
   var checkScenarioProperties = async function(data, mm, customer) {
-    //help.debug("checking scenario", JSON.stringify(data));
+    help.debug("checking scenario", data.marketMakerEthBalance);
 
     assert.equal(data.marketMakerMonth, await mm.getCurrentPeriodIndex());
-    data.marketMakerEthBalance.should.be.bignumber.equal(web3.eth.getBalance(mm.address));
+    data.marketMakerEthBalance.should.be.bignumber.equal(web3.eth.getBalance(mm.address), 5);
```

Comment:
why are we rounding here? Do you know where the difference comes from? Is it related to a change here?

In any case, please do it in a separate commit.

Reply:
It was WIP, I was working on teh last tests of market maker, the rounding is removed now :)